sys/auto_init/saul/auto_init_adcxx1c.c

```diff
@@ -42,19 +42,23 @@ static adcxx1c_t adcxx1c_devs[ADCXX1C_NUM];
  */
 static saul_reg_t saul_entries[ADCXX1C_NUM];
 
+/**
+ * @brief   Define the number of saul info
+ */
+#define ADCXX1C_INFO_NUM (sizeof(adcxx1c_saul_info) / sizeof(adcxx1c_saul_info[0]))
+
 /**
  * @brief   Reference the driver struct
  */
 extern saul_driver_t adcxx1c_saul_driver;
 
-
 void auto_init_adcxx1c(void)
 {
-    for (unsigned i = 0; i < ADCXX1C_NUM; i++) {
-        const adcxx1c_params_t *p = &adcxx1c_params[i];
+    assert(ADCXX1C_INFO_NUM == ADCXX1C_NUM);
```

Comment:
Since this is purely preprocessor based: is there a way to check this in the preprocessor instead? This way it is already clear at compile-time what's wrong. I'm thinking about something like:

```C
#ifdef ADCXX1C_INFO_NUM != ADCXX1C_NUM
#error "adcxx1c_params and adcxx1c_saul_info do not have the same number of elements"
#endif
```

Reply:
I'm not a preprocessor expert but are you sure that the preprocessor can detemine the actual values of `sizeof(adcxx1c_saul_info) / sizeof(adcxx1c_saul_info[0])` and `sizeof(adcxx1c_params) / sizeof(adcxx1c_params[0])` and be able to compare the results at compile time ?

Reply:
No, I'm not, but why not give it a try?

Reply:
(The main reason, why I gave it as a comment, not a change request is because I'm not sure)

Reply:
It doesn't work. As I said the preprocessor is just comparing strings or values and it's not able to revolve the C code behind an expression.

I have the following error:
`
auto_init_adcxx1c.c:48:33: error: missing binary operator before token "("
 #define ADCXX1C_INFO_NUM (sizeof(adcxx1c_saul_info) / sizeof(adcxx1c_saul_info[0]))
                                 ^
auto_init_adcxx1c.c:55:5: note: in expansion of macro 'ADCXX1C_INFO_NUM'
 #if ADCXX1C_INFO_NUM != ADCXX1C_NUM
`

I modified your suggestion as follows:
```c
#if ADCXX1C_INFO_NUM != ADCXX1C_NUM
#error "adcxx1c_params and adcxx1c_saul_info do not have the same number of elements"
#endif
```

Reply:
Alright then nevermind :-/